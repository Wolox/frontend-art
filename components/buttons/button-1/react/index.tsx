import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
