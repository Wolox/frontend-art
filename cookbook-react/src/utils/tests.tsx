import React, { ReactNode, useReducer } from 'react';
import { Switch, BrowserRouter as Router } from 'react-router-dom';

import { Context } from 'app/contexts/UserContext';
import { reducer, INITIAL_STATE } from 'app/contexts/UserContext/reducer';

interface Props {
  children: ReactNode;
}

function RootComponent({ children }: Props) {
  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);
  return (
    <Context.Provider value={{ state, dispatch }}>
      <Router>
        <Switch>{children}</Switch>
      </Router>
    </Context.Provider>
  );
}

export default RootComponent;
