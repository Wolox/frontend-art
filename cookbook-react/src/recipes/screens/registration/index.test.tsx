import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, fireEvent }  from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import 'mutationobserver-shim';

import { RootComponent } from '~utils/tests';

import Registration from '.';

jest.mock('i18next', () => ({
  t: (key: string) => key
}));

describe('#Registration', () => {
  const component = <Registration />;
  const validValues = { firstName: 'John', lastName: 'Doe', email: 'someEmail@wolox.com', password: 'aPassword' }

  describe('when mounting', () => {
    // it('shows valid content', () => {
    //   const instance = TestRenderer.create(component)
    //   expect(instance.toJSON()!).toMatchSnapshot();
    // })
  });

  describe('when values are empty and form is submitted', () => {
    it('shows the required message for each required field', async () => {
      const { getByRole, getByLabelText } = render(component);
      const firstName = await getByLabelText('Registration:firstName');
      const lastName = await getByLabelText('Registration:lastName');
      const email = await getByLabelText('Registration:email');
      const password = await getByLabelText('Registration:password');
      const passwordConfirmation = await getByLabelText('Registration:confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });
      
      await act(async () => {
        await fireEvent.submit(form);
      });

      expect(firstName.parentElement?.innerHTML).toMatch('Registration:requiredError');
      expect(lastName.parentElement?.innerHTML).toMatch('Registration:requiredError');
      expect(email.parentElement?.innerHTML).toMatch('Registration:requiredError');
      expect(password.parentElement?.innerHTML).toMatch('Registration:requiredError');
      expect(passwordConfirmation.parentElement?.innerHTML).toMatch('Registration:requiredError');
    })
  });

  describe('when email format is invalid and form is submitted', () => {
    it('shows the email error', async () => {
      const { getByRole, getByLabelText, getAllByRole } = render(component);
      const firstName = await getByLabelText('Registration:firstName');
      const lastName = await getByLabelText('Registration:lastName');
      const email = await getByLabelText('Registration:email');
      const password = await getByLabelText('Registration:password');
      const passwordConfirmation = await getByLabelText('Registration:confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });
      
      await act(async () => {
        await fireEvent.change(firstName, { target: { value: validValues.firstName } });
        await fireEvent.change(lastName, { target: { value: validValues.lastName } });
        await fireEvent.change(email, { target: { value: 'invalid email' } });
        await fireEvent.change(password, { target: { value: validValues.password } });
        await fireEvent.change(passwordConfirmation, { target: { value: validValues.password } });
        await fireEvent.submit(form);
      });
      const errors = await getAllByRole('alert');
  
      expect(email.parentElement?.innerHTML).toMatch('Registration:emailFormatError');
      expect(errors.length).toBe(1);
    })
  });

  describe('when password length is too short and form is submitted', () => {
    it('shows the password error', async () => {
      const { getByRole, getByLabelText, getAllByRole } = render(component);
      const firstName = await getByLabelText('Registration:firstName');
      const lastName = await getByLabelText('Registration:lastName');
      const email = await getByLabelText('Registration:email');
      const password = await getByLabelText('Registration:password');
      const passwordConfirmation = await getByLabelText('Registration:confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });
      
      await act(async () => {
        await fireEvent.change(firstName, { target: { value: validValues.firstName } });
        await fireEvent.change(lastName, { target: { value: validValues.lastName } });
        await fireEvent.change(email, { target: { value: validValues.email } });
        await fireEvent.change(password, { target: { value: 'short' } });
        await fireEvent.change(passwordConfirmation, { target: { value: 'short' } });
        await fireEvent.submit(form);
      });
      const errors = await getAllByRole('alert');

      expect(password.parentElement?.innerHTML).toMatch('Registration:passwordLengthError');
      expect(errors.length).toBe(1);
    })
  });

  describe('when password confirmation does not match password and form is submitted', () => {
    it('shows the password confirmation error', async () => {
      const { getByRole, getByLabelText, getAllByRole } = render(component);
      const firstName = await getByLabelText('Registration:firstName');
      const lastName = await getByLabelText('Registration:lastName');
      const email = await getByLabelText('Registration:email');
      const password = await getByLabelText('Registration:password');
      const passwordConfirmation = await getByLabelText('Registration:confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });
      
      await act(async () => {
        await fireEvent.change(firstName, { target: { value: validValues.firstName } });
        await fireEvent.change(lastName, { target: { value: validValues.lastName } });
        await fireEvent.change(email, { target: { value: validValues.email } });
        await fireEvent.change(password, { target: { value: validValues.password } });
        await fireEvent.change(passwordConfirmation, { target: { value: `${validValues.password}a` } });
        await fireEvent.submit(form);
      });
      const errors = await getAllByRole('alert');
      
      expect(passwordConfirmation.parentElement?.innerHTML).toMatch('Registration:confirmPasswordError');
      expect(errors.length).toBe(1);
    })
  });

  describe('when values are valid and form is submitted', () => {
    it('created the user', async () => {
      const { getByRole, getByLabelText, getAllByRole } = render(component);
      const firstName = await getByLabelText('Registration:firstName');
      const lastName = await getByLabelText('Registration:lastName');
      const email = await getByLabelText('Registration:email');
      const password = await getByLabelText('Registration:password');
      const passwordConfirmation = await getByLabelText('Registration:confirmPassword');
      const form = await getByRole('form', { name: 'registration-form' });
      
      await act(async () => {
        await fireEvent.change(firstName, { target: { value: validValues.firstName } });
        await fireEvent.change(lastName, { target: { value: validValues.lastName } });
        await fireEvent.change(email, { target: { value: validValues.email } });
        await fireEvent.change(password, { target: { value: validValues.password } });
        await fireEvent.change(passwordConfirmation, { target: { value: `${validValues.password}a` } });
        await fireEvent.submit(form);
      });
      const errors = await getAllByRole('alert');
      
      expect(passwordConfirmation.parentElement?.innerHTML).toMatch('Registration:confirmPasswordError');
      expect(errors.length).toBe(1);
    })
  });
});
