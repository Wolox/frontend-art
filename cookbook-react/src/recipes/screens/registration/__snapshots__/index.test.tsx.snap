// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Registration when mounting shows valid content 1`] = `
<form
  aria-label="registration-form"
  className="column center full-width formContainer"
  onSubmit={[Function]}
>
  <div
    className="column center m-bottom-3"
  >
    <h1
      className="m-bottom-1"
    >
      Registration:registration
    </h1>
    <h2>
      Registration:registrationExplanation
    </h2>
  </div>
  <div
    className="row space-between sectionContainer"
  >
    <div
      className="column start inputContainer item-1 m-right-2"
    >
      <label
        className=" m-bottom-1"
        htmlFor="firstName"
      >
        Registration:firstName
      </label>
      <input
        className="input input "
        disabled={false}
        id="firstName"
        name="firstName"
        placeholder="Registration:firstNamePlaceholder"
        readOnly={false}
        type="text"
      />
      <span
        aria-hidden={true}
        className=" errorText "
        id="firstName-error"
        role="alert"
      >
        
      </span>
    </div>
    <div
      className="column start inputContainer item-1"
    >
      <label
        className=" m-bottom-1"
        htmlFor="lastName"
      >
        Registration:lastName
      </label>
      <input
        className="input input "
        disabled={false}
        id="lastName"
        name="lastName"
        placeholder="Registration:lastNamePlaceholder"
        readOnly={false}
        type="text"
      />
      <span
        aria-hidden={true}
        className=" errorText "
        id="lastName-error"
        role="alert"
      >
        
      </span>
    </div>
  </div>
  <div
    className="row sectionContainer"
  >
    <div
      className="column start inputContainer item-1"
    >
      <label
        className=" m-bottom-1"
        htmlFor="email"
      >
        Registration:email
      </label>
      <input
        className="input input "
        disabled={false}
        id="email"
        name="email"
        placeholder="Registration:emailPlaceholder"
        readOnly={false}
        type="email"
      />
      <span
        aria-hidden={true}
        className=" errorText "
        id="email-error"
        role="alert"
      >
        
      </span>
    </div>
  </div>
  <div
    className="row space-between sectionContainer"
  >
    <div
      className="column start inputContainer item-1 m-right-2"
    >
      <label
        className=" m-bottom-1"
        htmlFor="password"
      >
        Registration:password
      </label>
      <input
        className="input input "
        disabled={false}
        id="password"
        name="password"
        placeholder="Registration:passwordPlaceholder"
        readOnly={false}
        type="password"
      />
      <span
        aria-hidden={true}
        className=" errorText "
        id="password-error"
        role="alert"
      >
        
      </span>
    </div>
    <div
      className="column start inputContainer item-1"
    >
      <label
        className=" m-bottom-1"
        htmlFor="confirmPassword"
      >
        Registration:confirmPassword
      </label>
      <input
        className="input input "
        disabled={false}
        id="confirmPassword"
        name="confirmPassword"
        placeholder="Registration:confirmPasswordPlaceholder"
        readOnly={false}
        type="password"
      />
      <span
        aria-hidden={true}
        className=" errorText "
        id="confirmPassword-error"
        role="alert"
      >
        
      </span>
    </div>
  </div>
  <div
    className="column center sectionContainer"
  >
    <button
      className="full-width m-bottom-1 button"
      type="submit"
    >
      RecoverPassword:enter
    </button>
    <a
      href="/login"
    >
      RecoverPassword:returnToLogin
    </a>
  </div>
</form>
`;
